src/shared/userdb.c

```diff
@@ -1605,13 +1608,16 @@ int groupdb_iterator_get(UserDBIterator *iterator, const UserDBMatch *match, Gro
                 if (r < 0)
                         return r;
 
-                if (gr && !group_record_match(gr, match))
+                r = group_record_match(gr, match);
+                if (r < 0)
```

Comment:
this would be bad, since it would mean if we iterate through the db and encounter a gid-less entry we'd abort the enumeration entirely, which would suck

Reply:
(See my comment above)